imageruler/test_imageruler.py

```diff
@@ -1,144 +1,154 @@
-import numpy as np
 import unittest
+import numpy as np
 import imageruler
-from imageruler.regular_shapes import disc, rounded_square, stripe
+try:
+    from imageruler.regular_shapes import disc, rounded_square, stripe
+except:
+    from regular_shapes import disc, rounded_square, stripe
+
 
+# properties of design weights array
 resolution = 1
 phys_size = (200, 200)
-diameters = np.linspace(2, 200, 11)
-
-
-def duality_dilation_erosion(pattern, diameter):
-    """
-    Test whether the duality relations between dilation and erosion are satisfied.
-
-    Args:
-        arr: A 2d array.
-        diameter: A float that represents the diameter of the kernel.
 
-    Returns:
-        A Boolean value that indicates whether the duality relations between erosion and dilation are satisfied.
-    """
-    erosion_of_negation = imageruler.binary_erode(~pattern, diameter)
-    negation_of_dilation = ~imageruler.binary_dilate(pattern, diameter)
-    dilation_of_negation = imageruler.binary_dilate(~pattern, diameter)
-    negation_of_erosion = ~imageruler.binary_erode(pattern, diameter)
-    return (erosion_of_negation
-            == negation_of_dilation).all() and (negation_of_erosion
-                                                == dilation_of_negation).all()
 
+class TestDuality(unittest.TestCase):
 
-def duality_closing_opening(pattern, diameter):
-    """
-    Test whether the duality relations between closing and opening are satisfied.
-
-    Args:
-        arr: A 2d array.
-        diameter: A float that represents the diameter of the kernel.
-
-    Returns:
-        A Boolean value that indicates whether the duality relations between closing and opening are satisfied.
-    """
-    opening_of_negation = imageruler.binary_open(~pattern, diameter)
-    negation_of_closing = ~imageruler.binary_close(pattern, diameter)
-    closing_of_negation = imageruler.binary_close(~pattern, diameter)
-    negation_of_opening = ~imageruler.binary_open(pattern, diameter)
-    return (opening_of_negation
-            == negation_of_closing).all() and (closing_of_negation
-                                               == negation_of_opening).all()
+    diameters = [2.2, 45.2, 50.1, 80.2, 101.5]
+
+    def duality_dilation_erosion(self,  weights: np.ndarray,
```

Comment:
These should be class methods or module functions since they use no state in `self`.

Reply:
Good point. Feel free to make a PR with the proposed changes.